src/org/rascalmpl/library/lang/html/IO.rsc

```diff
@@ -29,15 +29,15 @@ data HTMLSyntax
 This function uses [JSoup's](http://www.jsoup.org) HTML parser which is robust
 against errors in the HTML, and complete in the sense that it supports all of HTML.
 }
-java HTMLElement readHTMLFile(loc file, loc base=file);
+java HTMLElement readHTMLFile(loc file, loc base=file, bool trackOrigins=false, bool includeEndTags=true);
 
 @synopsis{Parse a HTML string and return an HTMLElement AST}
 @description{
 This function uses [JSoup's](http://www.jsoup.org) HTML parser which is robust
 against errors in the HTML, and complete in the sense that it supports all of HTML.
 }
 @javaClass{org.rascalmpl.library.lang.html.IO}
-java HTMLElement readHTMLString(str content, loc base=|http://localhost|);
+java HTMLElement readHTMLString(str content, loc base=|http://localhost|, bool trackOrigins=false, bool includeEndTags=true, loc src=|unknown:///|);
```

Comment:
what is the differeence between `base` and `src`?

Reply:
base is the HTML base URI for the content and src is the actual file that this HTML is in. `Base` is actually used by the jsoup parser for implementing features in HTML, while src is only used for the origin tracking.